### PR TITLE
Assignment 8

### PR DESCRIPTION
**Pull request by MeguminXIII**

### CODE REVIEW

**Comment by ZStuefen** [comment]
I will review this

**Comment by ZStuefen** [comment]
The unit tests did not work automatically because you named the new method Assign() rather than GenerateAssignments()
I changed the tests to test that method.

# Assignment
- Update the `GroupRepository` with functionality to generate gift assignments. ✔
  - A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
      - The error is created but Groups/Edit does not have the functionality to display it
  - A user is not allowed to be both the Giver and Recipient of the assignment. ✔
- Update the GroupsController with an endpoint for generating the assignments. ✔
- Update the Groups/Edit page to contain the UI elements for the following
  - A means to generate assignments. ❌
  - A means to display any errors that occurs while generating assignments. ❌
  - Display of the assignments for each of the users. ❌
      Groups/Edit was not changed
- Write appropriate unit tests for changes made to the `GroupRepository` ✔
- Write appropriate functional tests for the change made to the `GroupsController` ✔
    I'll check these off because the tests will still cover the Assign method if changed to use Assign() instead of GenerateAssignments(). One test does not pass due to a typo in "Groups must ahve at least 3 users"

## Extra Credit
- There is currently no way to for a user to add gifts. On the edit user page add functionality to be able to add a gift to the user. ❌
- Allow for navigating to a user from the Group edit page to make it easy to see the gifts requested by a user. ❌

File: SecretSanta/src/SecretSanta.Web/Views/Groups/Edit.cshtml [hunk 1]
```diff
@@ -40,5 +43,8 @@
         </select>
         <button class="border-1 bg-green-200 border-green-600 hover:bg-green-400 py-2 px-4 rounded-xl mr-3" @@click="addToGroup(group.id)">Add</button>
     </div>
+    <div class="py-4">
+        <button class="border-1 bg-green-400 border-black-800 hover:bg-red-600 py-2 px-4 rounded-xl mr-3" @@click="Assign(group.id)">Generate Assignments</button>
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Web/Views/Groups/Edit.cshtml:47]: The `Assign` method referenced here does not exist in the main.ts so clicking the button results in an error.

**Review by Keboo — COMMENTED** [review]
# Assignment
- Update the `GroupRepository` with functionality to generate gift assignments. ✔
  - A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
  - A user is not allowed to be both the Giver and Recipient of the assignment. ✔
- Update the GroupsController with an endpoint for generating the assignments. ✔
- Update the Groups/Edit page to contain the UI elements for the following
  - A means to generate assignments. ❌
  - A means to display any errors that occurs while generating assignments. ✔❌ - this is sort of there, but given the issue generating assignments it is difficult to test.
  - Display of the assignments for each of the users. ✔
- Write appropriate unit tests for changes made to the `GroupRepository` ✔
- Write appropriate functional tests for the change made to the `GroupsController` ✔

**Pull request closed without merging**